### ISSUE TITLE
Checklist Ordering and card counter broken when pasting a list

### ISSUE THREAD

**Issue by coursemyhorse** [RestyaPlatform/board#2047]
## Expected Behavior
Issue A) Checklist order should be always maintained and not randomly change. 
Issue B) The checklist counter shown on the card should reflect the true number of checklist items.

## Current Behavior
A: When pasting a list into the create checklist box to quickly build a multiple item checklist, issues occur after refreshing the card whereby the checklist order changes randomly depending on how many items you have. For example this simple checklist below each one named only as a number:

1
2
3
4
5

May become:

1
2
4
3
5

B: When pasting a list into the create checklist box to quickly build a multiple item checklist, issues occur after refreshing the card whereby the checklist counter on the main card shows an incorrect number of checklist items. Using again the above list with 5 items, after refreshing it can show 0/1 items completed, when it should show 0/5. If you then tick one of the 5 tasks, it can mark all of them as complete in green showing 1/1 tasks completed. On a larger scale in my testing I saw different numbers again:

Number pasted into create checklist box = 85
After refresh main card shows = 81

Number pasted into create checklist box = 81
After refresh main card shows = 73

Number pasted into create checklist box = 20
After refresh main card shows = 16

## Possible Solution
Review code to see how checklists are working when items are pasted in to the create box. 

## Steps to Reproduce (for bugs)
A)
1: Create a new card with any name
2: Drill into the card and click on the "checklist" button. The "Add checklist dialogue box should appear.
3: Leave the **Title** as the default name "Checklist" and leave the **Copy Items From** unchecked (i.e. do not copy from anywhere) and click the ADD button. This creates a new checklist on the card and defaults the cursor to be inside the checklist ready to type/paste items.
4: Paste the below list into dialogue box and either hit enter, or click ADD.

1
2
3
4
5

This should then build a multi item checklist with the above 5 items listed correctly in order initially. 

5: Either hit F5 on your keyboard, or refresh the page whilst staying in the card and you may see the checklist order change straight away. If not, exit the card and refresh the page whilst in the main board view, then go back into the card and the order will have changed.

![capture](https://user-images.githubusercontent.com/18658093/45230273-4a035f00-b2c0-11e8-94f8-895dcb4040bd.JPG)
B) Do the same as above steps to reproduce A, and after step 5, check the main checklist counter on the card and it will be incorrect.

![capture2](https://user-images.githubusercontent.com/18658093/45230591-3a384a80-b2c1-11e8-9619-ff5d114bd00c.jpg)

##Note
i: Issue A and B only occur when **PASTING** lists from your clipboard into the create checklist dialogue box. It does not occur if you manually create the above list by entering each item one at a time. 
ii: It is also possible to paste some lists which *do not* trigger the ordering issue A. In my testing, the below list did not trigger the issue and I do not know why:

Item 1
Item 2
Item 3
Item 4

And also this list did not trigger it either:

A My
B Pasted
C List

When this occurs however, the issue B (incorrect checklist counter) still occurs and is always wrong regardless. 

## Context
The ability to paste entire lists into the create checklist dialogue box to quickly build multi item checklists is an incredibly powerful feature, and one which many people want to use. We can quickly take deployment plans, runbooks, todo lists and copy paste them into restyaboard supporting markdown formatting to create nice checklists on cards. We can use this to collaborate in our team seeing in real time items being ticked off by anyone. So checklists are super important for me and I want to keep using the feature to create long lists without having to type each one out manually. The way each new line triggers a new item is fantastic and a feature we should keep.

## Your Environment
I am using the latest version 0.6.5 running on XAMPP on Windows Server 2012 using apache, php, postgres. I have done no customizations and only have admin and 2 users setup for testing. 2 boards and minimal cards.

**Comment by sridhar391** [comment]
The fix is available on the dev branch and it will be available on the upcoming release
